Source/JavaScriptCore/API/JSValueRef.h

```diff
@@ -292,6 +320,140 @@ JS_EXPORT JSValueRef JSValueMakeString(JSContextRef ctx, JSStringRef string);
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A uint32_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigUint64` and truncate it to a uint32_t if the numeric is a BigInt, otherwise do `ToUint32`.
+*/
+JS_EXPORT uint32_t JSValueToUInt32(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Converts a JSValue to a singed 32-bit integer and returns the resulting integer.
+    @param ctx        The execution context to use.
+    @param value      The JSValue to convert.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A int32_t with the result of conversion, or 0 if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first. Then apply `ToBigInt64` and truncate it to a int32_t if the numeric is a BigInt, otherwise do `ToInt32`.
+*/
+JS_EXPORT int32_t JSValueToInt32(JSContextRef ctx, JSValueRef value, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Compares a JSValue with an unsigned 64-bit integer.
+    @param ctx        The execution context to use.
+    @param left       The JSValue as the left operand.
+    @param right      The uint64_t as the right operand.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first and then compare with the integer.
+*/
+JS_EXPORT JSRelationCondition JSValueCompareUInt64(JSContextRef ctx, JSValueRef left, uint64_t right, JSValueRef* exception) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+    @function
+    @abstract         Compares a JSValue with a signed 64-bit integer.
+    @param ctx        The execution context to use.
+    @param left       The JSValue as the left operand.
+    @param right      The int64_t as the right operand.
+    @param exception  A pointer to a JSValueRef in which to store an exception, if any. Pass NULL if you do not care to store an exception.
+    @result           A value of JSRelationCondition, a kJSRelationConditionUndefined is returned if an exception is thrown.
+    @discussion       Convert the JSValue to a numeric by `ToNumeric` in JavaScript first and then compare with the integer.
```

Comment:
I would say "`left` is converted to an integer according to the rules specified by the JavaScript language then compared with `right`."